The code comes from the patch below however in 6.12 the rk3588s.dtsi file has been changed as per 
https://github.com/torvalds/linux/commit/def88eb4d8365a4aa064d28405d03550a9d0a3be
All the code seems to be split in "-base" and "-extra"so this patch is now applied to rk3588-base.dtsi 

The code is already in the Kernel but the HW is not enabled so we only need the DTS Patch
From 235ccbb0d859a5ff1ff6cc01daeb69effca933f8 Mon Sep 17 00:00:00 2001
From: Muhammed Efe Cetin <efectn@protonmail.com>
Date: Thu, 16 Nov 2023 17:52:35 +0300
Subject: [PATCH] arm64: dts: rockchip: add crypto node

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
index fc67585b64b7..5dcb420ca132 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
@@ -1879,6 +1879,16 @@ sdhci: mmc@fe2e0000 {
 		status = "disabled";
 	};
 
+    rng: rng@fe378000 {
+            compatible = "rockchip,trngv1";
+            reg = <0x0 0xfe378000 0x0 0x200>;
+            interrupts = <GIC_SPI 400 IRQ_TYPE_LEVEL_HIGH 0>;
+            clocks = <&scmi_clk SCMI_HCLK_SECURE_NS>;
+            clock-names = "hclk_trng";
+            resets = <&scmi_reset SRST_H_TRNG_NS>;
+            reset-names = "reset";
+    };
+	
 	i2s0_8ch: i2s@fe470000 {
 		compatible = "rockchip,rk3588-i2s-tdm";
 		reg = <0x0 0xfe470000 0x0 0x1000>;
